src/features/browser/BrowserWebViewScreen.tsx

```diff
@@ -270,39 +270,22 @@ const BrowserWebViewScreen = () => {
           return
         }
 
-        const signedTransactions = await Promise.all(
-          transactions.map(
-            async ({
-              transaction,
-            }: SolanaSignAndSendTransactionInput & {
-              transaction: Transaction | VersionedTransaction
-            }) => {
-              let signedTransaction:
-                | Transaction
-                | VersionedTransaction
-                | undefined
-              if (!isVersionedTransaction) {
-                // TODO: Verify when lookup table is needed
-                // transaction.add(lookupTableAddress)
-                signedTransaction =
-                  await anchorProvider?.wallet.signTransaction(
-                    transaction as Transaction,
-                  )
-              } else {
-                signedTransaction =
-                  await anchorProvider?.wallet.signTransaction(
-                    transaction as VersionedTransaction,
-                  )
-              }
-
-              if (!signedTransaction) {
-                throw new Error('Failed to sign transaction')
-              }
-
-              return signedTransaction
-            },
-          ),
-        )
+        const signedTransactions: (Transaction | VersionedTransaction)[] = []
+        // eslint-disable-next-line no-restricted-syntax
+        for (const txInput of transactions) {
+          try {
+            const { transaction } = txInput
+            const convertTx = isVersionedTransaction
+              ? (transaction as VersionedTransaction)
+              : (transaction as Transaction)
+            const signedTransaction =
+              await anchorProvider?.wallet.signTransaction(convertTx)
+            signedTransactions.push(signedTransaction)
+            await new Promise((resolve) => setTimeout(resolve, 2000))
```

Comment:
Why are we waiting 2 seconds?

Reply:
This issue arises when using Jupiter for swap transactions. Jupiter constructs two transactions, which means the cold wallet needs to sign twice. In terms of UI interaction, we need to open the modal twice in succession for QR code scanning and signing. However, a strange thing happens: while the first bottomSheetModalRef.current?.present() call successfully brings up the modal, the second call after completion fails to wake up the modal. So I added a 2-second sleep, after which the second modal was successfully awakened. I'm confused - is this caused by React Native performance issues or something else? I haven't been able to find the real reason for this problem. Can you help me?